hypershift-operator/controllers/nodepool/manifests.go

```diff
@@ -69,11 +69,50 @@ func TunedConfigMap(namespace, name string) *corev1.ConfigMap {
 	}
 }
 
-func PerformanceProfileConfigMap(namespace, name string) *corev1.ConfigMap {
+func PerformanceProfileConfigMap(namespace, name, nodePoolName string) *corev1.ConfigMap {
 	return &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: namespace,
-			Name:      "perfprofile-" + name,
+			Name:      composeValidName(name, nodePoolName),
 		},
 	}
 }
+
+// composeValidName takes two qualified objects` names
+// and composes a new qualified name out them by performing weighted prefixing
+func composeValidName(name1, name2 string) string {
```

Comment:
@stevekuznetsov Given the scope and the size of the function I suggest to move it into a central place and expose it as public, since it can be useful elsewhere.

Reply:
My personal rule is to do that sort of thing once the function needs to be re-used, instead of before-hand. In my opinion that helps to guard against premature optimizations or generalizations where we may be inferring possible future uses we don't know about yet. Once other users of the function exist, though, making it generic and a utility sounds good!

Reply:
I completely agree, but i'll need it here: https://github.com/openshift/hypershift/pull/4150#discussion_r1628125124

Reply:
If it's fine by you, we can keep this function in this file for now, and move it to central package on  the second PR. this way it would be easier to explain the justification.

Reply:
Sounds good to me!